pkg/ui/src/views/shared/components/alertMessage/alertMessage.tsx

```diff
@@ -0,0 +1,94 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+import React from "react";
+import { Alert, Icon } from "antd";
+import { Link } from "react-router-dom";
+
+import { AlertInfo, AlertLevel } from "src/redux/alerts";
+import "./alertMessage.styl";
+
+interface AlertMessageProps extends AlertInfo {
+  dismiss(): void;
+}
+
+type AlertType = "success" | "info" | "warning" | "error";
+
+const mapAlertLevelToType = (alertLevel: AlertLevel): AlertType => {
+  switch (alertLevel) {
+    case AlertLevel.SUCCESS:
+      return "success";
+    case AlertLevel.NOTIFICATION:
+      return "info";
+    case AlertLevel.WARNING:
+      return "warning";
+    case AlertLevel.CRITICAL:
+      return "error";
+    default:
+      return "info";
+  }
+};
+
+const getIconType = (alertLevel: AlertLevel): string => {
+  switch (alertLevel) {
+    case AlertLevel.SUCCESS:
+      return "check-circle";
+    case AlertLevel.NOTIFICATION:
+      return "info-circle";
+    case AlertLevel.WARNING:
+      return "warning";
+    case AlertLevel.CRITICAL:
+      return "close-circle";
+    default:
+      return "info-circle";
+  }
+};
+
+export class AlertMessage extends React.Component<AlertMessageProps> {
+  static defaultProps: Partial<AlertMessageProps> = {
```

Comment:
why is this defaultProps necessary?

Reply:
Indeed, not necessary! Has to be cleaned up.